src/Microsoft.DotNet.Interactive.HttpRequestParser/HttpLexer.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.DotNet.Interactive.HttpRequest;
+
+internal class HttpLexer
```

Comment:
Per our discussion on Tuesday, we decided to make the Lexer a private class inside the Parser (and always test the tokenization via the parser's public API).

Reply:
Resolving since this was done offline.